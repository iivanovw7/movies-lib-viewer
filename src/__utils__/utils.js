/**
 * Module contains test utils
 * @module __utils__/utils
 * @author Igor Ivanov
 */
import chalk from 'chalk';
import { mount, shallow } from 'enzyme';
import { createMemoryHistory } from 'history';
import React from 'react';
import { createStore } from 'redux';
import configureMockStore from 'redux-mock-store';
import createSagaMiddleware from 'redux-saga';
import { ThemeProvider } from 'styled-components';

import settings from '../config/data';
import rootReducer from '../store/configureStore';

/**
 * Theme mock value.
 * @type {{mode: string}}
 */
const theme = {
  mode: settings.theme,
};

/**
 * Configure saga middleware.
 */
const sagaMiddleware = createSagaMiddleware({});

/* eslint-disable */

/**
 * @param {object} initialState - initial state to store
 * @function storeFactory
 * @returns {Store} Redux store
 */
export const storeFactory = initialState => {
  return createStore(rootReducer, initialState);
};

//Browser history mock
export const history = createMemoryHistory('/');

//Store mock
export const mockStore = configureMockStore([sagaMiddleware]);

export const testName = (title, description) => `[${chalk.yellow(title)}]: ${description}`;

/**
 * Styled theme mount HOC
 * @param child
 * @return {*}
 */
export function mountWithTheme(child) {
  return mount(child, {
    wrappingComponent: ({ children }) => <ThemeProvider theme={theme}>{children}</ThemeProvider>,
  });
}

/**
 * Styled theme shallow HOC
 * @param child
 * @return {*}
 */
export function shallowWithTheme(child) {
  return shallow(child, {
    wrappingComponent: ({ children }) => <ThemeProvider theme={theme}>{children}</ThemeProvider>,
  });
}

/* eslint-enable */
