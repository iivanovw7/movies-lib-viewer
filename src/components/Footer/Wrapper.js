/**
 * Module contains application footer wrapper
 * @module components/Footer/Wrapper
 * @author Igor Ivanov
 */
import styled from 'styled-components';

import { base } from '../../config/styles';
import { zIndex } from '../../style/mixins';

const Wrapper = styled.footer`
  bottom: 0;
  display: flex;
  flex-direction: row;
  height: ${base.footerHeight}em;
  justify-content: center;
  position: fixed;
  width: 100%;
  z-index: ${zIndex.Footer};
`;

export default Wrapper;
