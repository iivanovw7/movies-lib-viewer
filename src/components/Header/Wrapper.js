/**
 * Module contains application header wrapper
 * @module components/Header/Wrapper
 * @author Igor Ivanov
 */
import styled from 'styled-components';

import { base } from '../../config/styles';
import { zIndex } from '../../style/mixins';
import { headerBg } from '../../style/theme/background';

const Wrapper = styled.header`
  background-color: ${headerBg};
  display: column;
  flex-direction: row;
  height: ${base.headerHeight}em;
  justify-content: center;
  margin-top: ${base.topBarHeight}em;
  position: fixed;
  top: 0;
  width: 100%;
  z-index: ${zIndex.Header};

  &::after {
    background: url(${(props) => props.bg});
    background-repeat: no-repeat;
    background-size: cover;
    bottom: 0;
    content: '';
    left: 0;
    opacity: 0.3;
    position: absolute;
    right: 0;
    top: 0;
    z-index: -1;
  }
`;

export default Wrapper;
