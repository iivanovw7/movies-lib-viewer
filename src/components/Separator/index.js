/**
 * Module contains Separator component
 * @module components/Separator
 * @author Igor Ivanov
 */
import { linearGradient } from 'polished';
import styled from 'styled-components';

import { colorSet } from '../../config/styles';
import { zIndex } from '../../style/mixins';

const { gradients } = colorSet;

const Separator = styled.div`
  height: ${(props) => props.height || 0.1}em;
  ${linearGradient({
    colorStops: [`${gradients.first} 0%`, `${gradients.second} 50%`, `${gradients.third} 95%`],
    toDirection: 'to right',
    fallback: gradients.first,
  })};
  width: 100%;
  z-index: ${zIndex.Separator};
`;

export default Separator;
