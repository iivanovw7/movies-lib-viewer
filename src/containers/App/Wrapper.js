/**
 * Module contains main application wrapper element
 * @module containers/App/Wrapper
 * @author Igor Ivanov
 */
import styled from 'styled-components';

import { mainBg } from '../../style/theme/background';

const Wrapper = styled.div`
  background-color: ${mainBg};
  margin: 0 auto;
  max-width: 100vw;
  min-height: 100vh;
  -ms-overflow-style: none;
  overflow-x: hidden;
  overflow-y: auto;
  padding: 0;
  scroll-behavior: smooth;
  scrollbar-width: none;
`;

export default Wrapper;
