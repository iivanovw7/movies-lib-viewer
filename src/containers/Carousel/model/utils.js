/**
 * Module contains carousel page utils
 * @module containers/Carousel/model/utils
 */

import setting from '../../../config/data';

/**
 * Calculates previous position.
 * @param {number} length - slides length.
 * @param {number} current - current slide.
 * @param {number} [skip = 0] - slide to be skipped.
 * @return {number} previous slide position.
 */
export function previous(length, current, skip = 0) {
  return Number((current - 1 - skip + length) % length);
}

/**
 * Calculates next position.
 * @param {number} length - slides length.
 * @param {number} current - current slide.
 * @param {number} [skip = 0] - slide to be skipped.
 * @return {number} next slide position.
 */
export function next(length, current, skip = 0) {
  return Number((current + 1 + skip) % length);
}

/**
 * Returns carousel element threshold.
 * @param {Object} target - target element.
 * @return {number} target threshold
 */
export function threshold(target) {
  const width = target.clientWidth;
  return width / 3;
}

/**
 * Carousel swipe event handler.
 * @param {Object} eventData data.
 * @param {number} length - slides length.
 * @param {number} dir - swipe direction.
 * @return {null | number} swipe length or null.
 */
export function swiped(eventData, length, dir = -1) {
  const t = threshold(eventData.event.target);
  const d = dir * eventData.deltaX;

  return d >= t ? length : null;
}

const basePosterUrl = (size) => `${setting.net.tmdbImageUrl}/t/p/w${size}/`;

export const posterUrl = (slide, size) => `${basePosterUrl(size)}${slide.poster_path}`;
