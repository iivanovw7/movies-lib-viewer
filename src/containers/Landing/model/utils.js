/**
 * Module contains landing page utils
 * @module containers/Landing/model/utils
 */
import { filter, pipe, prop } from 'ramda';

import { isString } from '../../../utils/helpers';

// eslint-disable-next-line import/prefer-default-export
export const withPosters = (list) => filter(pipe(prop('poster_path'), isString))(list);
