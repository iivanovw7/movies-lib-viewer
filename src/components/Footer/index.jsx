/**
 * Module contains application footer
 * @module components/Footer
 * @author Igor Ivanov
 */
import React from 'react';

import Separator from '../Separator';

import Wrapper from './Wrapper';

/**
 * Creates application footer component.
 * @method
 * @constructor
 * @return {Node} React component with children.
 */
function Footer() {
  return (
    <Wrapper>
      <Separator height={0.3} />
    </Wrapper>
  );
}

export default Footer;
