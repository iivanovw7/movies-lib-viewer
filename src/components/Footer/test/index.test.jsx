/**
 * Module contains Footer component test suite
 */
import { mount, shallow } from 'enzyme';
import React from 'react';

import { testName } from '../../../__utils__/utils';
import Separator from '../../Separator';
import Footer from '../index';
import Wrapper from '../Wrapper';

// prettier-ignore
describe(testName('Footer', 'component test suite'), function FooterTestSuite() {
  // eslint-disable-next-line require-jsdoc
  const Composition = (props, shouldMount = false) => {
    // eslint-disable-next-line react/jsx-props-no-spreading
    const node = <Footer {...props} />;

    return shouldMount ? mount(node) : shallow(node);
  };

  it('Should match snapshot', () => {
    const component = Composition({});

    expect(component)
      .toMatchSnapshot();
  });

  it('Should render wrapper', () => {
    const component = Composition({});

    expect(component.find(Wrapper))
      .toHaveLength(1);
  });

  it('Should render Separator', () => {
    const component = Composition({});

    expect(component.find(Separator))
      .toHaveLength(1);
  });
});
