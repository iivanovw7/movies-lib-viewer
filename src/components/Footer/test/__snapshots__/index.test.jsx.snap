// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[[33mFooter[39m]: component test suite Should match snapshot 1`] = `
<Wrapper>
  <Separator
    height={0.3}
  />
</Wrapper>
`;
